x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutput.java

```diff
@@ -158,6 +169,17 @@ public void writeCachedBlock(Block block) throws IOException {
         nextCachedBlock++;
     }
 
+    public Integer fromCache(FieldAttribute attr) {
+        return cachedFieldAttributes.get(attr);
+    }
+
+    public Integer addToCache(FieldAttribute attr) {
+        assert cachedFieldAttributes.containsKey(attr) == false;
```

Comment:
nit: we could throw IAE to make the error message more debug-friendly? Or return null if the FA was already in the cache?

Reply:
Let's go with an IAE